db/system_keyspace.cc

```diff
@@ -1751,6 +1754,70 @@ future<> set_bootstrap_state(bootstrap_state state) {
     });
 }
 
+class nodetool_status_table : public memtable_filling_virtual_table {
+public:
+    nodetool_status_table() : memtable_filling_virtual_table(build_schema()) {}
+
+    static schema_ptr build_schema() {
+        auto id = generate_legacy_id(NAME, "status");
+        return schema_builder(NAME, "status", std::make_optional(id))
+            .with_column("peer", inet_addr_type, column_kind::partition_key)
+            .with_column("dc", utf8_type)
+            .with_column("up", boolean_type)
+            .with_column("status", utf8_type)
+            .with_column("load", utf8_type)
+            .with_column("tokens", int32_type)
+            .with_column("owns", float_type)
+            .with_column("host_id", uuid_type)
+            .with_version(generate_schema_version(id))
+            .build();
+    }
+
+    future<> execute(memtable& mt) override {
```

Comment:
is this the memtable which the virtual table will be read from at the time the command is executed? is it cleared in between? or is the result cached in memtable itself?

Reply:
Every read gets a new memtable. On each query we first fill the memtable, then read from it. No caching.